great_expectations/data_context/data_context/data_context.py

```diff
@@ -363,13 +363,63 @@ def __init__(
             ge_cloud_config=ge_cloud_config,
         )
 
-        # save project config if data_context_id auto-generated or global config values applied
-        project_config_dict = dataContextConfigSchema.dump(project_config)
+        # Save project config if data_context_id auto-generated
+        if self._check_for_usage_stats_sync(project_config):
+            self._save_project_config()
+
+    def _check_for_usage_stats_sync(self, project_config: DataContextConfig) -> bool:
```

Comment:
Do we account for differences in how arrays/list items are ordered?

`["foo", "bar"] != ["bar", "foo"]`

I don't have enough context to know if we actually need to care about that.

Reply:
Also on subject of diffs.
This is a really great (and light-weight) pytest-plugin for visualizing diffs.
https://github.com/hjwp/pytest-icdiff

Reply:
ordering shouldn't matter here! and I loveeee that pytest plugin haha good call!